src/test/java/com/firebolt/jdbc/resultset/FireboltResultSetTest.java

```diff
@@ -1053,21 +1056,124 @@ void shouldReturnDataForNewNonNumericDataTypes() throws SQLException {
 		assertNull(resultSet.getObject(13));
 	}
 
+	@SuppressWarnings("java:S5961") // all assertions are necessary
```

Comment:
They are necessary, but the SC issue here is with having a huge single "god" test. I'd rather not suppress thing needlessly and refactor this into two (or more) test cases. One I can think of is having all errors test, e.g. `shouldErrorOnTooLargeNumericTypes`. Or even separating it into `shouldGetObjectsForByte`, `shouldGetObjectsForShort` , etc. if the previous suggestion is not enough.

Reply:
Done. Now we have separate functions for each type.